Multiple bugs with gitlab bridge.

Hi,

Playing with git-bug on a existing gitlab repository with a few existing issues on the vendor's bug tracker,  I ran trough the following errors:

After having successfully configured the bridge I could import the gitlab issues as well as creating a couple of local issues with `git bug add`, so far so good. Then, I meet the following errors:
* when doing again `git bug bridge pull`, issues are duplicated with new hashes. 
* nothing is exported when doing `git bug bridge push`, it jus returns `exported 0 issues with myBridge bridge`.

I've tested the bridge on a freshly created repo on gitlab, again the first steps goes well, but this time when doing `git bug bridge pull` a second time to test if duplicates arise, I got this crash:

```
panic: runtime error: index out of range [-1]                                                                                                        
                                                                        
goroutine 51 [running]:       
github.com/MichaelMure/git-bug/bridge/gitlab/iterator.(*issueIterator).Value(...)
        /home/dtrckd/src/_sink/git-bug/bridge/gitlab/iterator/issue.go:38
github.com/MichaelMure/git-bug/bridge/gitlab/iterator.(*Iterator).NextIssue(0xc0010ba000, 0xc000043dd0)
        /home/dtrckd/src/_sink/git-bug/bridge/gitlab/iterator/iterator.go:86 +0x244
github.com/MichaelMure/git-bug/bridge/gitlab.(*gitlabImporter).ImportAll.func1(0xc000260100, 0xc0000aed00, 0xc0010bc000)
        /home/dtrckd/src/_sink/git-bug/bridge/gitlab/import.go:70 +0x547
created by github.com/MichaelMure/git-bug/bridge/gitlab.(*gitlabImporter).ImportAll
        /home/dtrckd/src/_sink/git-bug/bridge/gitlab/import.go:66 +0x146

```
Similar scenario on Github seems to work fine.

Version:
git-bug version: 0.7.1-dev-28c3ee3b71
github.com/xanzy/go-gitlab v0.29.0
go version go1.14.1 linux/amd64

Gitlab: 12.9.2 (self hosted)
OS: Linux 4.19.0-8-amd64 / Debian 10 x86_64

I've received this error as well. Create an identity locally, then followed these steps:

1. `git bug bridge configure` - set up gitlab as the bridge
2. 
```
git bug bridge push
new issue: e80238b17c133382dc01ba74eef316ad50d84e5c
exported 1 issues with gitlab bridge
```
3. Made a change in the Gitlab UI (applied a label to the issue).
4.
```
git bug bridge pull
new identity: 7cef8e713b2299d8a8035d038c42d5c29e75cda9
new issue: 0d4041eb258226f712c9391835e7faf409792dc9
imported 1 issues and 1 identities with gitlab bridge
```
5. `git bug termui` - notice that there are now two issues listed, one with the label, one without.
6. 
```
git bug bridge pull
panic: runtime error: index out of range [-1]

goroutine 57 [running]:
github.com/MichaelMure/git-bug/bridge/gitlab/iterator.(*issueIterator).Value(...)
	/home/travis/gopath/src/github.com/MichaelMure/git-bug/bridge/gitlab/iterator/issue.go:38
github.com/MichaelMure/git-bug/bridge/gitlab/iterator.(*Iterator).NextIssue(0xc000296000, 0x0)
	/home/travis/gopath/src/github.com/MichaelMure/git-bug/bridge/gitlab/iterator/iterator.go:86 +0x244
github.com/MichaelMure/git-bug/bridge/gitlab.(*gitlabImporter).ImportAll.func1(0xc0002541c0, 0xc000142d80, 0xc00008c1e0)
	/home/travis/gopath/src/github.com/MichaelMure/git-bug/bridge/gitlab/import.go:70 +0x560
created by github.com/MichaelMure/git-bug/bridge/gitlab.(*gitlabImporter).ImportAll
	/home/travis/gopath/src/github.com/MichaelMure/git-bug/bridge/gitlab/import.go:66 +0x146
```


Performing a pull from within `git bug termui` "works," in that `done` is reported.

Hi, I am receiving the same `panic: runtime error: index out of range [-1]` on the repo that I used for testing in #360.

For reference:
* I used git-bug 0.7.0 on https://gitlab.com/mkroehnert/git-bug-issue-test/-/issues (which had the label issue).
* I ran `git bug bridge pull` with git-bug 0.7.1 on the same checkout that held the initial `git bug bridge pull` from the above mentioned ticket.
* The exact error received after the bridge pull is (same as in the very first message):
```
panic: runtime error: index out of range [-1]

goroutine 23 [running]:
github.com/MichaelMure/git-bug/bridge/gitlab/iterator.(*issueIterator).Value(...)
        /build/source/bridge/gitlab/iterator/issue.go:38
github.com/MichaelMure/git-bug/bridge/gitlab/iterator.(*Iterator).NextIssue(0xc0000ae180, 0x0)
        /build/source/bridge/gitlab/iterator/iterator.go:86 +0x244
github.com/MichaelMure/git-bug/bridge/gitlab.(*gitlabImporter).ImportAll.func1(0xc0002161c0, 0xc0000aed00, 0xc00021c5a0)
        /build/source/bridge/gitlab/import.go:70 +0x547
created by github.com/MichaelMure/git-bug/bridge/gitlab.(*gitlabImporter).ImportAll
        /build/source/bridge/gitlab/import.go:66 +0x146
```

--- a/bridge/gitlab/import.go
+++ b/bridge/gitlab/import.go
@@ -127,8 +127,8 @@ func (gi *gitlabImporter) ensureIssue(repo *cache.RepoCache, issue *gitlab.Issue
 	b, err := repo.ResolveBugMatcher(func(excerpt *cache.BugExcerpt) bool {
 		return excerpt.CreateMetadata[core.MetaKeyOrigin] == target &&
 			excerpt.CreateMetadata[metaKeyGitlabId] == parseID(issue.IID) &&
-			excerpt.CreateMetadata[metaKeyGitlabBaseUrl] == gi.conf[confKeyProjectID] &&
-			excerpt.CreateMetadata[metaKeyGitlabProject] == gi.conf[confKeyGitlabBaseUrl]
+			excerpt.CreateMetadata[metaKeyGitlabBaseUrl] == gi.conf[confKeyGitlabBaseUrl] &&
+			excerpt.CreateMetadata[metaKeyGitlabProject] == gi.conf[confKeyProjectID]
 	})
 	if err == nil {
 		return b, nil

